lib/matplotlib/backend_bases.py

```diff
@@ -3460,7 +3500,11 @@ def new_figure_manager_given_figure(cls, num, figure):
 
     @classmethod
     def draw_if_interactive(cls):
-        if cls.mainloop is not None and is_interactive():
+        manager_class = cls.FigureCanvas.manager_class
+        interactive_backend = (
```

Comment:
```suggestion
        backend_is_interactive = (
            # interactive backends need to overrwrite one of these methods
```

should we maybe have a `FigureManagerBase.has_interactive_backend()` that provides this functionality?

Reply:
Changed the variable name.  I'd rather not add a method as the relevant customization point has been changed by #23291 anyways.